File: src/main/scala/com/kapilkoju/autopos/transaction/api/PaymentApi.scala
package com.kapilkoju.autopos.transaction.api

import com.kapilkoju.autopos.kernel.api.BaseApi
import com.kapilkoju.autopos.transaction.domain.Payment
import com.kapilkoju.autopos.transaction.repo.PaymentRepo
import org.springframework.web.bind.annotation.{RequestMapping, RestController}

@RestController
@RequestMapping(Array("/api/payments"))
class PaymentApi(private val paymentRepo: PaymentRepo)
  extends BaseApi[Payment](paymentRepo, "payment", "/api/payments")

File: src/main/kotlin/com/kapilkoju/autopos/transaction/api/PaymentApi.kt
package com.kapilkoju.autopos.transaction.api

import com.kapilkoju.autopos.kernel.api.BaseApi
import com.kapilkoju.autopos.transaction.domain.Payment
import com.kapilkoju.autopos.transaction.service.PaymentRepo
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(PaymentApi.baseUrl)
class PaymentApi(paymentRepo: PaymentRepo)
    : BaseApi<Payment>(paymentRepo, "payment", baseUrl) {

    companion object {
        const val baseUrl = "/api/payments"
    }
}
